### PR TITLE
[digiplex] Introduce trouble reports (#7117)

### PR DESCRIPTION
**Pull request by rmichalak**
Introduces handling of Trouble/Trouble Restore events from PRT3 ASCII protocol.

### CODE REVIEW

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/events/TroubleEvent.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.openhab.binding.digiplex.internal.communication.DigiplexMessageHandler;
+
+/**
+ * Message providing global trouble status
+ *
+ * @author Robert Michalak - Initial contribution
+ *
+ */
+public class TroubleEvent extends AbstractEvent {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/events/TroubleEvent.java]: Please add `@NonNullByDefault` to all of your classes

File: bundles/org.openhab.binding.digiplex/src/main/resources/ESH-INF/thing/bridge.xml [hunk 2]
```diff
@@ -62,4 +86,106 @@
...
+				<option value="ON">Failure</option>
+				<option value="OFF">OK</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="global_fire_loop">
+		<item-type>Switch</item-type>
+		<label>Global Fire Loop</label>
+		<description>Reports if fire Loop has been triggered</description>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/resources/ESH-INF/thing/bridge.xml]:
```suggestion
		<description>Reports if fire loop has been triggered</description>
```

Comment (rmichalak) [reply bundles/org.openhab.binding.digiplex/src/main/resources/ESH-INF/thing/bridge.xml]: done

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/events/TroubleEvent.java (hunk 1 above)
Comment (rmichalak) [reply bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/events/TroubleEvent.java]: I filled the gaps in all classes, not only those strictly related to this new feature. This is why I put them in a separate commit. Not that I think it makes sense in all cases, as some of them contain only primitive types, but rules are rules... ;)

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/AreaLabelResponse.java [hunk 3]
```diff
@@ -12,15 +12,20 @@
...
  *
  * @author Robert Michalak - Initial contribution
  *
  */
+@NonNullByDefault
 public class AreaLabelResponse extends AbstractResponse {
 
     public final int areaNo;
+    @Nullable
     public final String areaName;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/AreaLabelResponse.java]:
```suggestion
    public final @Nullable String areaName;
```

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/AreaStatusResponse.java [hunk 4]
```diff
@@ -12,14 +12,19 @@
...
  * Response for {@link AreaStatusRequest}
  *
  * @author Robert Michalak - Initial contribution
  */
+@NonNullByDefault
 public class AreaStatusResponse extends AbstractResponse {
 
     public final int areaNo;
+    @Nullable
     public final AreaStatus status;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/AreaStatusResponse.java]:
```suggestion
    public final @Nullable AreaStatus status;
```

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneLabelResponse.java [hunk 5]
```diff
@@ -12,15 +12,20 @@
...
  *
  * @author Robert Michalak - Initial contribution
  *
  */
+@NonNullByDefault
 public class ZoneLabelResponse extends AbstractResponse {
 
     public final int zoneNo;
+    @Nullable
     public final String zoneName;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneLabelResponse.java]:
```suggestion
    public final @Nullable String zoneName;
```

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneStatusResponse.java [hunk 6]
```diff
@@ -12,15 +12,20 @@
...
  *
  * @author Robert Michalak - Initial contribution
  *
  */
+@NonNullByDefault
 public class ZoneStatusResponse extends AbstractResponse {
 
     public final int zoneNo;
+    @Nullable
     public final ZoneStatus status;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneStatusResponse.java]:
```suggestion
    public final @Nullable ZoneStatus status;
```

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/handler/DigiplexBridgeHandler.java [hunk 7]
```diff
@@ -255,6 +259,16 @@ public void handleCommunicationStatus(CommunicationStatus response) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
             }
         }
+
+        @Override
+        public void handleTroubleEvent(TroubleEvent troubleEvent) {
+
```
Comment (cpmeister) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/handler/DigiplexBridgeHandler.java]: This blank line bothers me. Please remove it. 😄

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneStatusResponse.java (hunk 6 above)
Comment (rmichalak) [reply bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/communication/ZoneStatusResponse.java]: Again, separate commit for this change, as I made it binding-wide, to keep the same layout in every file.

**Pull request merged**